### PR TITLE
MGMT-7250: Generate discovery ignition on demand

### PR DESCRIPTION
**Pull request by djzager**
# Assisted Pull Request

## Description

Generate the discovery ignition, when requested, on demand.

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [x] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @carbonin 
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -820,6 +820,7 @@ func (b *bareMetalInventory) updateImageInfoPostUpload(ctx context.Context, infr
 				return errors.New("Failed to generate image: error generating URL")
 			}
 		} else {
+			// TODO(djzager): Needs to be updated with image-service work, MGMT-3934
```
Comment (carbonin) [inline internal/bminventory/inventory.go:823]: Do you want to commit this?

File: internal/cluster/cluster.go [hunk 2]
```diff
@@ -51,7 +51,6 @@ var S3FileNames = []string{
 	"kubeadmin-password",
 	"kubeconfig-noingress",
 	"install-config.yaml",
-	"discovery.ign",
```
Comment (carbonin) [inline internal/cluster/cluster.go:54]: Is there an enum in the swagger somewhere for this as well?

Comment (djzager) [reply internal/cluster/cluster.go:54]: there is. But we still want requests to `/clusters/$CLUSTER_ID/downloads/files?file_name=discovery.ign` to work...right?

Comment (carbonin) [reply internal/cluster/cluster.go:54]: Yup, right, my bad :sweat_smile:

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (djzager) [reply internal/bminventory/inventory.go:823]: I think I'll remove this. Will wait to see if I have any issues verifying this manually.

**Comment by djzager** [comment]
# Before this change

via `v1/clusters/$CLUSTER_ID/downloads/files?file_name=discovery.ign`:

```
{
  "ignition": {
    "version": "3.1.0"
  },
  "passwd": {
    "users": [
      {"groups":["sudo"],"name":"core","passwordHash":"!","sshAuthorizedKeys":["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCnAVtn9Bomee4fTG7Ydr1Ne3Nb+BpcP9T/jeu7b7LgF6Q4Tk4kwaLaUPpLXoDXt5KJ3eK8FUdaVCBKVSYfkJqxD1k/Lvqjzley/1XwFMqNNWCoN0JRwBtGmcXjhnYH6eMzkzr0OSy0FAudKwwa2eXtu/C58uuUN0vC632rfDdposPGta/EU8OXhmeNY38dIt7Tb8IWbOvOQ0S4XLbAYLSOVLWkf6shex1GWiSSAeE8J3fq8rfwgACW/hXHQRqeJ5UNz6PTK1xTXuEi74zcAkbXbwVB+v4gncfKNxc36DsAu7h6HzWTzeWswmf19V8/hF/NXcTrfLsZifIfrIJXVVFa/D57xSK9PEEGzlohaEOqo9PVv4nPLz3w8iQbPs/akqDxxpLsZ2i9AP4oUawEeXlT1q7bVyY7g2XwapWM+UQXq3UZJXF6IlVccL3u7yk1hwkEzwwKENLI/Y970gVZGQ1u9bKyCSTh9OQhvqcsYsYBthO7ZY0cIBjq3Kas9TYmjXk= dev@edge-04.edge.lab.eng.rdu2.redhat.com"]}
    ]
  },
  "systemd": {
    "units": [{
      "name": "agent.service",
      "enabled": true,
      "contents": "[Service]\nType=simple\nRestart=always\nRestartSec=3\nStartLimitInterval=0\nEnvironment=HTTP_PROXY=\nEnvironment=http_proxy=\nEnvironment=HTTPS_PROXY=\nEnvironment=https_proxy=\nEnvironment=NO_PROXY=\nEnvironment=no_proxy=\nTimeoutStartSec=180\nExecStartPre=/usr/local/bin/agent-fix-bz1964591 quay.io/ocpmetal/assisted-installer-agent:latest\nExecStartPre=podman run --privileged --rm -v /usr/local/bin:/hostbin quay.io/ocpmetal/assisted-installer-agent:latest cp /usr/bin/agent /hostbin\nExecStart=/usr/local/bin/agent --url http://<pending>:8090 --cluster-id 6aae6063-5250-4e75-96f3-d3e9af16f33f --agent-version quay.io/ocpmetal/assisted-installer-agent:latest --insecure=true  \n\n[Unit]\nWants=network-online.target\nAfter=network-online.target\n\n[Install]\nWantedBy=multi-user.target"
    },
    {
        "name": "selinux.service",
        "enabled": true,
        "contents": "[Service]\nType=oneshot\nExecStartPre=checkmodule -M -m -o /root/assisted.mod /root/assisted.te\nExecStartPre=semodule_package -o /root/assisted.pp -m /root/assisted.mod\nExecStart=semodule -i /root/assisted.pp\n\n[Install]\nWantedBy=multi-user.target"
    }
    ]
  },
  "storage": {
    "files": [{
      "overwrite": true,
      "path": "/usr/local/bin/agent-fix-bz1964591",
      "mode": 755,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%23%21%2Fusr%2Fbin%2Fsh%0A%0A%23%20This%20script%20is%20a%20workaround%20for%20bugzilla%201964591%20where%20symlinks%20inside%20%2Fvar%2Flib%2Fcontainers%2F%20get%0A%23%20corrupted%20under%20some%20circumstances.%0A%23%0A%23%20In%20order%20to%20let%20agent.service%20start%20correctly%20we%20are%20checking%20here%20whether%20the%20requested%0A%23%20container%20image%20exists%20and%20in%20case%20%22podman%20images%22%20returns%20an%20error%20we%20try%20removing%20the%20faulty%0A%23%20image.%0A%23%0A%23%20In%20such%20a%20scenario%20agent.service%20will%20detect%20the%20image%20is%20not%20present%20and%20pull%20it%20again.%20In%20case%0A%23%20the%20image%20is%20present%20and%20can%20be%20detected%20correctly%2C%20no%20any%20action%20is%20required.%0A%0AIMAGE=$%28echo%20$1%20%7C%20sed%20%27s%2F:.%2A%2F%2F%27%29%0Apodman%20images%20%7C%20grep%20$IMAGE%20%7C%7C%20podman%20rmi%20--force%20$1%20%7C%7C%20true%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/motd",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0AThis%20is%20a%20host%20being%20installed%20by%20the%20OpenShift%20Assisted%20Installer.%0AIt%20will%20be%20installed%20from%20scratch%20during%20the%20installation.%0A%0AThe%20primary%20service%20is%20agent.service.%20To%20watch%20its%20status%2C%20run:%0Asudo%20journalctl%20-u%20agent.service%0A%0ATo%20view%20the%20agent%20log%2C%20run:%0Asudo%20journalctl%20TAG=agent%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/NetworkManager/conf.d/01-ipv6.conf",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%5Bconnection%5D%0Aipv6.dhcp-iaid=mac%0Aipv6.dhcp-duid=ll%0A" }
    },
    {
        "overwrite": true,
        "path": "/root/.docker/config.json",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "..." }
    },
    {
        "overwrite": true,
        "path": "/root/assisted.te",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "data:text/plain;base64,...=" }
    }]
  }
}% 
```

`v1/clusters/$CLUSTER_ID/discovery-ignition`:
```
{"config":"{\n  \"ignition\": {\n    \"version\": \"3.1.0\"\n  },\n  \"passwd\": {\n    \"users\": [\n      {\"groups\":[\"sudo\"],\"name\":\"core\",\"passwordHash\":\"!\",\"sshAuthorizedKeys\":[\"..."]}\n    ]\n  },\n  \"systemd\": {\n    \"units\": [{\n      \"name\": \"agent.service\",\n      \"enabled\": true,\n      \"contents\": \"[Service]\\nType=simple\\nRestart=always\\nRestartSec=3\\nStartLimitInterval=0\\nEnvironment=HTTP_PROXY=\\nEnvironment=http_proxy=\\nEnvironment=HTTPS_PROXY=\\nEnvironment=https_proxy=\\nEnvironment=NO_PROXY=\\nEnvironment=no_proxy=\\nTimeoutStartSec=180\\nExecStartPre=/usr/local/bin/agent-fix-bz1964591 quay.io/ocpmetal/assisted-installer-agent:latest\\nExecStartPre=podman run --privileged --rm -v /usr/local/bin:/hostbin quay.io/ocpmetal/assisted-installer-agent:latest cp /usr/bin/agent /hostbin\\nExecStart=/usr/local/bin/agent --url http://<pending>:8090 --cluster-id 6aae6063-5250-4e75-96f3-d3e9af16f33f --agent-version quay.io/ocpmetal/assisted-installer-agent:latest --insecure=true  \\n\\n[Unit]\\nWants=network-online.target\\nAfter=network-online.target\\n\\n[Install]\\nWantedBy=multi-user.target\"\n    },\n    {\n        \"name\": \"selinux.service\",\n        \"enabled\": true,\n        \"contents\": \"[Service]\\nType=oneshot\\nExecStartPre=checkmodule -M -m -o /root/assisted.mod /root/assisted.te\\nExecStartPre=semodule_package -o /root/assisted.pp -m /root/assisted.mod\\nExecStart=semodule -i /root/assisted.pp\\n\\n[Install]\\nWantedBy=multi-user.target\"\n    }\n    ]\n  },\n  \"storage\": {\n    \"files\": [{\n      \"overwrite\": true,\n      \"path\": \"/usr/local/bin/agent-fix-bz1964591\",\n      \"mode\": 755,\n      \"user\": {\n          \"name\": \"root\"\n      },\n      \"contents\": { \"source\": \"data:,%23%21%2Fusr%2Fbin%2Fsh%0A%0A%23%20This%20script%20is%20a%20workaround%20for%20bugzilla%201964591%20where%20symlinks%20inside%20%2Fvar%2Flib%2Fcontainers%2F%20get%0A%23%20corrupted%20under%20some%20circumstances.%0A%23%0A%23%20In%20order%20to%20let%20agent.service%20start%20correctly%20we%20are%20checking%20here%20whether%20the%20requested%0A%23%20container%20image%20exists%20and%20in%20case%20%22podman%20images%22%20returns%20an%20error%20we%20try%20removing%20the%20faulty%0A%23%20image.%0A%23%0A%23%20In%20such%20a%20scenario%20agent.service%20will%20detect%20the%20image%20is%20not%20present%20and%20pull%20it%20again.%20In%20case%0A%23%20the%20image%20is%20present%20and%20can%20be%20detected%20correctly%2C%20no%20any%20action%20is%20required.%0A%0AIMAGE=$%28echo%20$1%20%7C%20sed%20%27s%2F:.%2A%2F%2F%27%29%0Apodman%20images%20%7C%20grep%20$IMAGE%20%7C%7C%20podman%20rmi%20--force%20$1%20%7C%7C%20true%0A\" }\n    },\n    {\n      \"overwrite\": true,\n      \"path\": \"/etc/motd\",\n      \"mode\": 420,\n      \"user\": {\n          \"name\": \"root\"\n      },\n      \"contents\": { \"source\": \"data:,%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0AThis%20is%20a%20host%20being%20installed%20by%20the%20OpenShift%20Assisted%20Installer.%0AIt%20will%20be%20installed%20from%20scratch%20during%20the%20installation.%0A%0AThe%20primary%20service%20is%20agent.service.%20To%20watch%20its%20status%2C%20run:%0Asudo%20journalctl%20-u%20agent.service%0A%0ATo%20view%20the%20agent%20log%2C%20run:%0Asudo%20journalctl%20TAG=agent%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0A\" }\n    },\n    {\n      \"overwrite\": true,\n      \"path\": \"/etc/NetworkManager/conf.d/01-ipv6.conf\",\n      \"mode\": 420,\n      \"user\": {\n          \"name\": \"root\"\n      },\n      \"contents\": { \"source\": \"data:,%0A%5Bconnection%5D%0Aipv6.dhcp-iaid=mac%0Aipv6.dhcp-duid=ll%0A\" }\n    },\n    {\n        \"overwrite\": true,\n        \"path\": \"/root/.docker/config.json\",\n        \"mode\": 420,\n        \"user\": {\n            \"name\": \"root\"\n        },\n        \"contents\": { \"source\": \"data:...\" }\n    },\n    {\n        \"overwrite\": true,\n        \"path\": \"/root/assisted.te\",\n        \"mode\": 420,\n        \"user\": {\n            \"name\": \"root\"\n        },\n        \"contents\": { \"source\": \"data:text/plain;base64,...=\" }\n    }]\n  }\n}"}
```

After this change, both endpoints get a response similar to `v1/clusters/$CLUSTER_ID/discovery-ignition`. That is, everything wrapped up in a `.config`.

**Comment by carbonin** [comment]
I think we should keep the API consistent before and after this change.

**Comment by djzager** [comment]
Addressed the API inconsistency in https://github.com/openshift/assisted-service/pull/2267#issuecomment-888477046 .

```
$ curl "$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install/v1/clusters/$CLUSTER_ID/downloads/files?file_name=discovery.ign"
{
  "ignition": {
    "version": "3.1.0"
  },
  "passwd": {
    "users": [
      {"groups":["sudo"],"name":"core","passwordHash":"!","sshAuthorizedKeys":["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCnAVtn9Bomee4fTG7Ydr1Ne3Nb+BpcP9T/jeu7b7LgF6Q4Tk4kwaLaUPpLXoDXt5KJ3eK8FUdaVCBKVSYfkJqxD1k/Lvqjzley/1XwFMqNNWCoN0JRwBtGmcXjhnYH6eMzkzr0OSy0FAudKwwa2eXtu/C58uuUN0vC632rfDdposPGta/EU8OXhmeNY38dIt7Tb8IWbOvOQ0S4XLbAYLSOVLWkf6shex1GWiSSAeE8J3fq8rfwgACW/hXHQRqeJ5UNz6PTK1xTXuEi74zcAkbXbwVB+v4gncfKNxc36DsAu7h6HzWTzeWswmf19V8/hF/NXcTrfLsZifIfrIJXVVFa/D57xSK9PEEGzlohaEOqo9PVv4nPLz3w8iQbPs/akqDxxpLsZ2i9AP4oUawEeXlT1q7bVyY7g2XwapWM+UQXq3UZJXF6IlVccL3u7yk1hwkEzwwKENLI/Y970gVZGQ1u9bKyCSTh9OQhvqcsYsYBthO7ZY0cIBjq3Kas9TYmjXk= dev@edge-04.edge.lab.eng.rdu2.redhat.com"]}
    ]
  },
  "systemd": {
    "units": [{
      "name": "agent.service",
      "enabled": true,
      "contents": "[Service]\nType=simple\nRestart=always\nRestartSec=3\nStartLimitInterval=0\nEnvironment=HTTP_PROXY=\nEnvironment=http_proxy=\nEnvironment=HTTPS_PROXY=\nEnvironment=https_proxy=\nEnvironment=NO_PROXY=\nEnvironment=no_proxy=\nTimeoutStartSec=180\nExecStartPre=/usr/local/bin/agent-fix-bz1964591 quay.io/ocpmetal/assisted-installer-agent:latest\nExecStartPre=podman run --privileged --rm -v /usr/local/bin:/hostbin quay.io/ocpmetal/assisted-installer-agent:latest cp /usr/bin/agent /hostbin\nExecStart=/usr/local/bin/agent --url http://<pending>:8090 --cluster-id 6aae6063-5250-4e75-96f3-d3e9af16f33f --agent-version quay.io/ocpmetal/assisted-installer-agent:latest --insecure=true  \n\n[Unit]\nWants=network-online.target\nAfter=network-online.target\n\n[Install]\nWantedBy=multi-user.target"
    },
    {
        "name": "selinux.service",
        "enabled": true,
        "contents": "[Service]\nType=oneshot\nExecStartPre=checkmodule -M -m -o /root/assisted.mod /root/assisted.te\nExecStartPre=semodule_package -o /root/assisted.pp -m /root/assisted.mod\nExecStart=semodule -i /root/assisted.pp\n\n[Install]\nWantedBy=multi-user.target"
    }
    ]
  },
  "storage": {
    "files": [{
      "overwrite": true,
      "path": "/usr/local/bin/agent-fix-bz1964591",
      "mode": 755,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%23%21%2Fusr%2Fbin%2Fsh%0A%0A%23%20This%20script%20is%20a%20workaround%20for%20bugzilla%201964591%20where%20symlinks%20inside%20%2Fvar%2Flib%2Fcontainers%2F%20get%0A%23%20corrupted%20under%20some%20circumstances.%0A%23%0A%23%20In%20order%20to%20let%20agent.service%20start%20correctly%20we%20are%20checking%20here%20whether%20the%20requested%0A%23%20container%20image%20exists%20and%20in%20case%20%22podman%20images%22%20returns%20an%20error%20we%20try%20removing%20the%20faulty%0A%23%20image.%0A%23%0A%23%20In%20such%20a%20scenario%20agent.service%20will%20detect%20the%20image%20is%20not%20present%20and%20pull%20it%20again.%20In%20case%0A%23%20the%20image%20is%20present%20and%20can%20be%20detected%20correctly%2C%20no%20any%20action%20is%20required.%0A%0AIMAGE=$%28echo%20$1%20%7C%20sed%20%27s%2F:.%2A%2F%2F%27%29%0Apodman%20images%20%7C%20grep%20$IMAGE%20%7C%7C%20podman%20rmi%20--force%20$1%20%7C%7C%20true%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/motd",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0AThis%20is%20a%20host%20being%20installed%20by%20the%20OpenShift%20Assisted%20Installer.%0AIt%20will%20be%20installed%20from%20scratch%20during%20the%20installation.%0A%0AThe%20primary%20service%20is%20agent.service.%20To%20watch%20its%20status%2C%20run:%0Asudo%20journalctl%20-u%20agent.service%0A%0ATo%20view%20the%20agent%20log%2C%20run:%0Asudo%20journalctl%20TAG=agent%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/NetworkManager/conf.d/01-ipv6.conf",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%5Bconnection%5D%0Aipv6.dhcp-iaid=mac%0Aipv6.dhcp-duid=ll%0A" }
    },
    {
        "overwrite": true,
        "path": "/root/.docker/config.json",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "..." }
    },
    {
        "overwrite": true,
        "path": "/root/assisted.te",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "data:text/plain;base64,...=" }
    }]
  }
}% 
```

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -3657,11 +3653,28 @@ func (b *bareMetalInventory) GetPresignedForClusterFiles(ctx context.Context, pa
...
+			return common.GenerateErrorResponder(err)
+		}
+
+		cfg, err := b.IgnitionBuilder.FormatDiscoveryIgnitionFile(infraEnv, b.IgnitionConfig, false, b.authHandler.AuthType())
+		if err != nil {
+			log.WithError(err).Error("Failed to format ignition config")
+			return common.GenerateErrorResponder(err)
+		}
+
+		return installer.NewDownloadClusterFilesOK().WithPayload(ioutil.NopCloser(strings.NewReader(cfg)))
```
Comment (carbonin) [inline internal/bminventory/inventory.go:3670]: I think we should also continue to use the `filemiddleware` here.

Comment (djzager) [reply internal/bminventory/inventory.go]: Fixed

```
$ curl -v "$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install/v1/clusters/$CLUSTER_ID/downloads/files?file_name=discovery.ign"
*   Trying 192.168.39.219:31426...
* Connected to 192.168.39.219 (192.168.39.219) port 31426 (#0)
> GET /api/assisted-install/v1/clusters/6aae6063-5250-4e75-96f3-d3e9af16f33f/downloads/files?file_name=discovery.ign HTTP/1.1
> Host: 192.168.39.219:31426
> User-Agent: curl/7.71.1
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Disposition: attachment; filename="discovery.ign"
< Content-Length: 8370
< Content-Type: application/octet-stream
< Date: Wed, 28 Jul 2021 21:29:30 GMT
<
{
  "ignition": {
    "version": "3.1.0"
  },
  "passwd": {
    "users": [
      {"groups":["sudo"],"name":"core","passwordHash":"!","sshAuthorizedKeys":["..."]}
    ]
  },
  "systemd": {
    "units": [{
      "name": "agent.service",
      "enabled": true,
      "contents": "[Service]\nType=simple\nRestart=always\nRestartSec=3\nStartLimitInterval=0\nEnvironment=HTTP_PROXY=\nEnvironment=http_proxy=\nEnvironment=HTTPS_PROXY=\nEnvironment=https_proxy=\nEnvironment=NO_PROXY=\nEnvironment=no_proxy=\nTimeoutStartSec=180\nExecStartPre=/usr/local/bin/agent-fix-bz1964591 quay.io/ocpmetal/assisted-installer-agent:latest\nExecStartPre=podman run --privileged --rm -v /usr/local/bin:/hostbin quay.io/ocpmetal/assisted-installer-agent:latest cp /usr/bin/agent /hostbin\nExecStart=/usr/local/bin/agent --url http://<pending>:8090 --cluster-id 6aae6063-5250-4e75-96f3-d3e9af16f33f --agent-version quay.io/ocpmetal/assisted-installer-agent:latest --insecure=true  \n\n[Unit]\nWants=network-online.target\nAfter=network-online.target\n\n[Install]\nWantedBy=multi-user.target"
    },
    {
        "name": "selinux.service",
        "enabled": true,
        "contents": "[Service]\nType=oneshot\nExecStartPre=checkmodule -M -m -o /root/assisted.mod /root/assisted.te\nExecStartPre=semodule_package -o /root/assisted.pp -m /root/assisted.mod\nExecStart=semodule -i /root/assisted.pp\n\n[Install]\nWantedBy=multi-user.target"
    }
    ]
  },
  "storage": {
    "files": [{
      "overwrite": true,
      "path": "/usr/local/bin/agent-fix-bz1964591",
      "mode": 755,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%23%21%2Fusr%2Fbin%2Fsh%0A%0A%23%20This%20script%20is%20a%20workaround%20for%20bugzilla%201964591%20where%20symlinks%20inside%20%2Fvar%2Flib%2Fcontainers%2F%20get%0A%23%20corrupted%20under%20some%20circumstances.%0A%23%0A%23%20In%20order%20to%20let%20agent.service%20start%20correctly%20we%20are%20checking%20here%20whether%20the%20requested%0A%23%20container%20image%20exists%20and%20in%20case%20%22podman%20images%22%20returns%20an%20error%20we%20try%20removing%20the%20faulty%0A%23%20image.%0A%23%0A%23%20In%20such%20a%20scenario%20agent.service%20will%20detect%20the%20image%20is%20not%20present%20and%20pull%20it%20again.%20In%20case%0A%23%20the%20image%20is%20present%20and%20can%20be%20detected%20correctly%2C%20no%20any%20action%20is%20required.%0A%0AIMAGE=$%28echo%20$1%20%7C%20sed%20%27s%2F:.%2A%2F%2F%27%29%0Apodman%20images%20%7C%20grep%20$IMAGE%20%7C%7C%20podman%20rmi%20--force%20$1%20%7C%7C%20true%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/motd",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0AThis%20is%20a%20host%20being%20installed%20by%20the%20OpenShift%20Assisted%20Installer.%0AIt%20will%20be%20installed%20from%20scratch%20during%20the%20installation.%0A%0AThe%20primary%20service%20is%20agent.service.%20To%20watch%20its%20status%2C%20run:%0Asudo%20journalctl%20-u%20agent.service%0A%0ATo%20view%20the%20agent%20log%2C%20run:%0Asudo%20journalctl%20TAG=agent%0A%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%20%20%2A%2A%0A" }
    },
    {
      "overwrite": true,
      "path": "/etc/NetworkManager/conf.d/01-ipv6.conf",
      "mode": 420,
      "user": {
          "name": "root"
      },
      "contents": { "source": "data:,%0A%5Bconnection%5D%0Aipv6.dhcp-iaid=mac%0Aipv6.dhcp-duid=ll%0A" }
    },
    {
        "overwrite": true,
        "path": "/root/.docker/config.json",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "..." }
    },
    {
        "overwrite": true,
        "path": "/root/assisted.te",
        "mode": 420,
        "user": {
            "name": "root"
        },
        "contents": { "source": "data:text/plain;base64,Cm1vZHVsZSBhc3Npc3RlZCAxLjA7CnJlcXVpcmUgewogICAgICAgIHR5cGUgY2hyb255ZF90OwogICAgICAgIHR5cGUgY29udGFpbmVyX2ZpbGVfdDsKICAgICAgICB0eXBlIHNwY190OwogICAgICAgIGNsYXNzIHVuaXhfZGdyYW1fc29ja2V0IHNlbmR0bzsKICAgICAgICBjbGFzcyBkaXIgc2VhcmNoOwogICAgICAgIGNsYXNzIHNvY2tfZmlsZSB3cml0ZTsKfQojPT09PT09PT09PT09PSBjaHJvbnlkX3QgPT09PT09PT09PT09PT0KYWxsb3cgY2hyb255ZF90IGNvbnRhaW5lcl9maWxlX3Q6ZGlyIHNlYXJjaDsKYWxsb3cgY2hyb255ZF90IGNvbnRhaW5lcl9maWxlX3Q6c29ja19maWxlIHdyaXRlOwphbGxvdyBjaHJvbnlkX3Qgc3BjX3Q6dW5peF9kZ3JhbV9zb2NrZXQgc2VuZHRvOwo=" }
    }]
  }
* Connection #0 to host 192.168.39.219 left intact
}%
```

**Review by carbonin — APPROVED** [review]

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (djzager) [reply internal/bminventory/inventory.go:824]: Decided to leave it so I can find it when it comes time to address MGMT-3934

**Pull request merged**